### ISSUE TITLE
ValueError: Big-endian buffer not supported on little-endian compiler

### ISSUE THREAD

**Issue by yas-nakajima** [quatrope/astroalign#70]
`reg_im2, footprint = astroalign.register(imdata2, imdata1)` fails with an error message of  
'ValueError: Big-endian buffer not supported on little-endian compiler' 
when using scikit-image=0.16.2 or 0.18.1. 
However, it works with scikit-image=0.15.0, so I downgraded scikit-image to 0.15.0 to get around the problem. 

I guess this is the same issue as the following. 
https://github.com/scikit-image/scikit-image/issues/4525

Using astroalign-2.4 and python3.7 

Thanks.

**Comment by martinberoiz** [comment]
Hello,

Yes, the issue is similar to the one in scikit-image issue you mentioned. The endian-ness from the source of your data doesn't match the endian-ness of your local computer where you're running astroaling.

Fortunately Numpy lets you swap the endianness quite easy:

https://numpy.org/doc/stable/user/basics.byteswapping.html

    target = target.newbyteorder()
    source = source.newbyteorder()

I don't think astroalign should deal with different endian issues but the above workaround works well.
Let me know if that fixes the issue for you.

**Comment by yas-nakajima** [comment]
The program without  .newbyteorder() lines works for the same FITS files with scikit-image==0.15.0. 
The FITS files does not contain `nan`s and , for example, DS9 displays the FITS file properly. 

It seems .newbyteorder() destroyed data. 
```
target = fits.getdata('target.fits')
source = fits.getdata('source.fits')
print('original')
print(np.max(target), np.min(target)) 
print(np.max(source), np.min(source))

target_new = target.newbyteorder()
source_new = source.newbyteorder()

print('newbyte')
print(np.max(target_new), np.min(target_new)) 
print(np.max(source_new), np.min(source_new))

```
original 
38990.938 345.18137
39516.312 -2829.9492
newbyte
nan nan
nan nan

**Comment by martinberoiz** [comment]
That is strange. Would you mind sending me the problem images so I can take a look? Thanks.

**Comment by yas-nakajima** [comment]
```
target_fixed = np.array(target, dtype="<f4")
source_fixed = np.array(source, dtype="<f4")
```

is a workaround.

**Comment by yas-nakajima** [comment]
```
- target = target.newbyteorder()
- source = source.newbyteorder()
+ target = target.byteswap().newbyteorder()
+ source = source.byteswap().newbyteorder()
```
FYI.  byteswap() is needed. 

Because skimage.transform.warp does not accept Big-endian,  
I would suggest adding .byteswap().newbyteorder() in  `def apply_transform` or `def _data` of astroalign.py.

**Comment by martinberoiz** [comment]
Yes, this is a known issue that was introduced in a recent version of scikit-image.

**Comment by Jayesh6999** [comment]
#Allgignment of Images

image_path='path name'
filename='image name'
files=sorted(glob.glob(os.path.join(image_path,filename)))
print(len(files))

ref_image=fits.open(files[0], )
reference_image=ref_image[0].data

for i in range(0,len(files)):
    image_data=fits.open(files[i])
    source_image=image_data[0].data + 0 # here the addition of zero (0) solves the the endian compiler issue. basically just add 
                                                                      zero to the image / file data
    header=image_data[0].header
    image_aligned,footprint=aa.register(source_image,reference_image)

    aligned_file=files[i].replace('.fits','')
    fits.writeto(aligned_file+'_aligned'+'.fits',image_aligned,header,overwrite=True)
    
    print('No. %i alignment done'%i)

# i hope this helps

**Comment by Jayesh6999** [comment]
#Allgignment of Images
image_path='image path'
filename='image / file name'
files=sorted(glob.glob(os.path.join(image_path,filename)))
print(len(files))

ref_image=fits.open(files[0], )
reference_image=ref_image[0].data

for i in range(0,len(files)):
    image_data=fits.open(files[i])
    source_image=image_data[0].data + 0  #here the addition of zero (0) solves the the endian compiler issue.
    header=image_data[0].header
    image_aligned,footprint=aa.register(source_image,reference_image)
    aligned_file=files[i].replace('.fits','')
    fits.writeto(aligned_file+'_aligned'+'.fits',image_aligned,header,overwrite=True)
    print('No. %i alignment done'%i)

(after for loop the rest of the code is in indentation)
